src/Workspaces/CSharp/Portable/Classification/Worker_Preprocesser.cs

```diff
@@ -263,6 +263,17 @@ private void ClassifyPragmaWarningDirective(PragmaWarningDirectiveTriviaSyntax n
             AddClassification(node.WarningKeyword, ClassificationTypeNames.PreprocessorKeyword);
             AddClassification(node.DisableOrRestoreKeyword, ClassificationTypeNames.PreprocessorKeyword);
 
+            // Comments after a "#pragma warning" directive without any error codes are trailing trivia of the "disable"/"restore" token.
```

Comment:
❓ Why not just call `ClassifyDirectiveTrivia` at the end of this method, like that seen in other cases:

https://github.com/dotnet/roslyn/blob/8c0679915d947048efa5786fc7f1f9b1bf1efdcc/src/Workspaces/CSharp/Portable/Classification/Worker_Preprocesser.cs#L288

Reply:
That makes two tests fail (namely `PP_PragmaWarning[Disable|Restore]OneWithComment`), because the actual output contains one additional `Comment("//Goo")`. Those two tests have error codes in the directive, so the classifier function calls `ClassifyNodeOrToken` on the error code node, from where it will eventually reach `ClassifyTriviaList` and `ClassifyTrivia`, adding a second classification for the comment. And we can't pass `allowComments: false` because that would defeat the entire purpose of that call.

Reply:
💭 Would it work to call `ClassifyDirectiveTrivia` only if `node.ErrorCodes` is empty?

Reply:
Yes, that works. Thanks! I've updated the PR.

Reply:
🎉